src/webapp/components/search-input/SearchInput.tsx

```diff
@@ -0,0 +1,115 @@
+import React, { useCallback, useEffect, useState } from "react";
+import { TextField } from "@material-ui/core";
+import { IconSearch24 } from "@dhis2/ui";
+import styled from "styled-components";
+
+import i18n from "../../../utils/i18n";
+
+type SearchInputProps = {
+    value: string;
+    onChange: (event: string) => void;
+    placeholder?: string;
+    disabled?: boolean;
+};
+
+export const SearchInput: React.FC<SearchInputProps> = React.memo(
+    ({ value, onChange, placeholder = "", disabled = false }) => {
+        const [stateValue, updateStateValue] = useState(value);
+
+        useEffect(() => updateStateValue(value), [value]);
+
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+        const onChangeDebounced = useCallback(
+            debounce((value: string) => {
+                if (onChange) onChange(value);
+            }, 400),
+            [onChange]
+        );
```

Comment:
Most of the times we should not disable eslint, but find a refactor. This works?

```
const onChangeDebounced = React.useMemo(
            () =>
                debounce((value: string) => {
                    if (onChange) onChange(value);
                }, 400),
            [onChange]
        );
```

Reply:
It works thanks!!